controllers/configurationpolicy_controller.go

```diff
@@ -195,7 +195,14 @@ func (r *ConfigurationPolicyReconciler) PeriodicallyExecConfigPolicies(
 			_ = r.refreshDiscoveryInfo()
 		}
 
-		if discoveryErr == nil {
+		cleanupImmediately, err := r.cleanupImmediately()
+		if err != nil {
+			log.Error(err, "Failed to determine if it's time to cleanup immediately")
+
+			skipLoop = true
+		}
+
+		if discoveryErr == nil || cleanupImmediately {
```

Comment:
in cleanupImmediately() we have 
```
	return false, fmt.Errorf(
		"isBeingUninstalled error: '%v', definitionIsDeleting error: '%v'", beingUninstalledErr, defErr,
	)
```
so... is it ok?

Reply:
Yes, because the "happy path" returns with:
```
	if beingUninstalledErr == nil && defErr == nil {
		// if either was deleting, we would've already returned.
		return false, nil
	}
```